### PR TITLE
mgr/cephadm: --no-overwite flag for apply command

### PR DESCRIPTION
**Pull request by adk3798**
no-overwrite flag makes it so cephadm will only apply the spec
if the spec does not already exist.

Signed-off-by: Adam King <adking@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/pybind/mgr/orchestrator/module.py [hunk 1]
```diff
@@ -328,7 +328,7 @@ def _add_host(self, hostname: str, addr: Optional[str] = None, labels: Optional[
         _status = 'maintenance' if maintenance else ''
         s = HostSpec(hostname=hostname, addr=addr, labels=labels, status=_status)
 
-        return self._apply_misc([s], False, Format.plain)
+        return self._apply_misc([s], False, Format.plain, False)
```
Comment (liewegas) [inline src/pybind/mgr/orchestrator/module.py:331]: since the default is False this isn't needed, right?

**Comment by liewegas** [comment]
Looks good to me!  Want to add a patch that includes ``no_overwrite=True`` for mgr/volumes?

**Review by jmolmo — APPROVED** [review]

**Comment by adk3798** [comment]
> Looks good to me! Want to add a patch that includes `no_overwrite=True` for mgr/volumes?

added

**Review by liewegas — APPROVED** [review]

**Comment by tchaikov** [comment]
- https://pulpito.ceph.com/kchai-2021-03-27_12:17:59-rados-wip-kefu-testing-2021-03-27-1545-distro-basic-gibba/
- https://pulpito.ceph.com/kchai-2021-03-28_03:01:25-rados-wip-kefu-testing-2021-03-27-1545-distro-basic-smithi/

failures

- https://tracker.ceph.com/issues/49344
- https://tracker.ceph.com/issues/49260
- https://tracker.ceph.com/issues/49633

**Pull request merged**